text/0048-move-replayid-out-of-tags.md

```diff
@@ -0,0 +1,63 @@
+- Start Date: 2023-01-05
+- RFC Type: decision
+- RFC PR: https://github.com/getsentry/rfcs/pull/48
+- RFC Status: draft
+
+# Summary
+
+This RFC is almost exactly the same as https://github.com/getsentry/rfcs/pull/47.
+
+Right now replay_ids are added to the error/transaction event tags object in the [event schema](https://sentry.io/organizations/sentry/performance/summary/replays/?project=11276&query=&statsPeriod=24h&transaction=%2Fsettings%2Faccount%2Femails%2F). This isn't ideal as tags should be reserved for user use only.
+
+We'd like to create a context object `replay` which for now would contain the singular field `replay_id`.
+
+We also want to move this potentially out of the tags columns within clickhouse for errors/txs. The replay_id can be stored as its own column or as a key in contexts.
+
+# Motivation
+
+Replays are linked together with errors and transactions throughout the product, relying on the current tag setup. We want to do this in a non-janky future-proof way.
+
+# Options Considered
+
+## SDK
+
+### Option A: Add a new context type
+
+```json
+{
+  ...,
+  "contexts": {
+    ...,
+    "replay": {
+      "replay_id": "<id>"
+    }
+  }
+}
+```
+
+### Other Options
+
+I suppose we could add the "replay_id" top level to contexts. Not sure what other options there could be.
+
+## Storage
```

Comment:
The difference between A and B (I don't think C is a good idea) will depend on how you plan to query on this column. Can you give some examples of how the column will be used?

Reply:
common use cases:

1. from a list of issue_ids, select and return the event_id and replay_id if a replay_id exists on the event
2. from a list of transaction_names, select and return the event_id and replay_id if a replay_id exists on the event
3. from a replay_id, fetch all associated issues
4. from a replay_id, fetch associated transactions 
5. select the column as a default field in the performance page, and on discover, for errors/txs

Reply:
My take on this would be to suggest option B then, based on queries 3 and 4. If you are filtering by replay ID having a dedicated column would be best. It's also what we recommended to the profiling team. @nikhars do you have any other thoughts?

Reply:
The `replay_id` column seems to be a loose linking column between two different datasets. I would also recommend a dedicated column based on its usage. So I would agree with Evan.

Reply:
got it, thanks!